jbpm-console-ng-rest/src/main/java/org/jbpm/console/ng/rest/ProcessImageResourceImpl.java

```diff
@@ -0,0 +1,183 @@
+        // Get kjar and see if svg file is present in it.
+        String [] depIdParts = deploymentId.split(":");
+        File depFile = repository.getArtifactFileFromRepository(new GAV(depIdParts[0], depIdParts[1], depIdParts[2]));
+        if( depFile != null ) { 
+            try {
+                JarFile depJarFile = new JarFile(depFile);
+                Enumeration<JarEntry> entries = depJarFile.entries();
+                while( entries.hasMoreElements() ) { 
+                    JarEntry jarEntry = entries.nextElement();
+                    if( jarEntry.getName().contains(svgFileName) ) { 
+                        InputStream svgInputStream = depJarFile.getInputStream(jarEntry);
+
+                        procDefSvg = IOUtils.toString(svgInputStream);
+                        break;
+                    }
+                }
+            } catch( IOException e ) {
+                // no-op: could not retrieve file from deployment
+            }
+        }
+        
+        return procDefSvg;
+    }
+    
+    // Rest methods --------------------------------------------------------------------------------------------------------------
+
+    @GET
+    @Path("/image")
+//    @Produces({MediaType.APPLICATION_SVG_XML, MediaType.APPLICATION_OCTET_STREAM})
+    public Response getProcessImage(  @PathParam("deploymentId") String deploymentId, @PathParam("processDefId" ) String processId) {
+       
+        // find procdef (or throw 404) if the deployment id or pro
+        ProcessDefinition procDef = dataService.getProcessesByDeploymentIdProcessId(deploymentId, processId);
+        if( procDef == null ) { 
+           return Response.status(Response.Status.NOT_FOUND).build();
+        }
+       
+        // get SVG String
+        String imageSVGString = getProcesImageSVGFromDeployment(deploymentId, procDef);
+        if( imageSVGString == null ) { 
+            return Response.status(Response.Status.PRECONDITION_FAILED).build();
+        } 
+            
+        return Response.ok(imageSVGString, MediaType.APPLICATION_SVG_XML_TYPE).build();
+    }
+    
+
+    @GET
+    @Path("/image/{procInstId: [0-9]+}")
+    public Response getActiveProcessImage(  @PathParam("deploymentId") String deploymentId, @PathParam("processDefId" ) String processId, 
+            @PathParam("procInstId") long procInstId) {
+       
+        ProcessDefinition procDef = dataService.getProcessesByDeploymentIdProcessId(deploymentId, processId);
+        if( procDef == null ) { 
+           return Response.status(Response.Status.NOT_FOUND).build();
+        }
+                
+        // get SVG String
+        String imageSVGString = getProcesImageSVGFromDeployment(deploymentId, procDef);
+        if( imageSVGString == null ) { 
+            return Response.status(Response.Status.PRECONDITION_FAILED).build();
+        }
+       
+        // find active nodes and modify image
+        List<NodeInstanceLog> logs = getAuditLogService().findNodeInstances(procInstId);
```

Comment:
no need to do two calls, you can use one to get both active and completed nodes: https://github.com/droolsjbpm/jbpm/blob/master/jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/RuntimeDataService.java#L201